includes/controllers/class.llms.controller.certificates.php

```diff
@@ -131,6 +133,118 @@ public function maybe_handle_reporting_actions() {
 
 	}
 
+	/**
+	 * Handle awrded certificates sync actions.
+	 *
+	 * @since [version]
+	 *
+	 * @return void|WP_Error
+	 */
+	public static function maybe_handle_awarded_certificates_sync_actions() {
+
+		if ( ! llms_verify_nonce( '_llms_cert_sync_actions_nonce', 'llms-cert-sync-actions', 'GET' ) ) {
+			return new WP_Error(
+				'llms-sync-awarded-certificates-nonce',
+				__( 'Sorry, you are not allowed to do this', 'lifterlms' )
+			);
+		}
+
+		if ( ! isset( $_GET['action'] ) ) {
+			return new WP_Error(
+				'llms-sync-awarded-certificates-missing-action',
+				__( 'Sorry, you have not provided any actions', 'lifterlms' )
+			);
+		}
+
+		switch ( $_GET['action'] ) {
+			case 'sync_awarded_certificate':
+				$error  = new WP_Error(
+					'llms-sync-awarded-certificate-missing-certificate-id',
+					__( 'Sorry, you need to provide a valid awarded certificate ID.', 'lifterlms' )
+				);
+				$method = 'sync_awarded_certificate';
+				break;
+			case 'sync_awarded_certificates':
+				$error  = new WP_Error(
+					'llms-sync-awarded-certificates-missing-template-id',
+					__( 'Sorry, you need to provide a valid certificate template ID.', 'lifterlms' )
+				);
+				$method = 'sync_awarded_certificates';
+				break;
+			default:
+				return new WP_Error(
+					'llms-sync-awarded-certificates-invalid-action',
+					__( 'You\'re trying to perform an invalid action.', 'lifterlms' )
+				);
+		}
+
+		$cert_id = llms_filter_input( INPUT_GET, 'post', FILTER_SANITIZE_NUMBER_INT );
+		if ( empty( $cert_id ) ) {
+			return $error;
+		}
+
+		return self::{$method}( $cert_id );
+
+	}
+
+	/**
+	 * Sync awarded certificate with its template.
+	 *
+	 * @since [version]
+	 *
+	 * @param int $cert_id Awarded certificate id.
+	 * @return void|WP_Error
+	 */
+	private static function sync_awarded_certificate( $cert_id ) {
+
+		if ( ! current_user_can( get_post_type_object( 'llms_my_certificate' )->cap->edit_post, $cert_id ) ) {
```

Comment:
```suggestion
		if ( ! current_user_can( 'edit_post', $cert_id ) ) {
```

Doesn't WP take care of figuring out what the actual cap is automatically based on the id passed in?

Reply:
I guess you're right, we're end up doing this, since `map_meta_cap` is `false` for awarded certificates.
https://github.com/WordPress/wordpress-develop/blob/5.8.1/src/wp-includes/capabilities.php#L171
I guess I have this habit of specify that that way... dunno.